core-implementation/src/main/java/gg/solarmc/loader/impl/login/NameAddressHistoryUpdate.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * DataLoader
+ * Copyright © 2021 SolarMC Developers
+ *
+ * DataLoader is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * DataLoader is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with DataLoader. If not, see <https://www.gnu.org/licenses/>
+ * and navigate to version 3 of the GNU Affero General Public License.
+ */
+
+package gg.solarmc.loader.impl.login;
+
+import gg.solarmc.loader.Transaction;
+import gg.solarmc.loader.impl.UserDetails;
+import org.jooq.DSLContext;
+
+import static gg.solarmc.loader.schema.tables.LibertybansAddresses.LIBERTYBANS_ADDRESSES;
+import static gg.solarmc.loader.schema.tables.LibertybansNames.LIBERTYBANS_NAMES;
+
+interface NameAddressHistoryUpdate {
+
+    void update(Transaction transaction, UserDetails userDetails);
+
+    class NoOpImpl implements NameAddressHistoryUpdate {
+
+        @Override
+        public void update(Transaction transaction, UserDetails userDetails) {
+
+        }
+    }
+
+    class FunctioningImpl implements NameAddressHistoryUpdate {
```

Comment:
Does dataloader depend on libertybans now? I know ariel said that we're gonna be using it, and it's a good thing this is interface-based

Reply:
The issue is, we don't want to duplicate name and address history storage. If libertybans were specifically made for solarmc, then the name and address history tables would be in data loader, and libertybans would therefore depend on the data loader tables.

However, using a public plugin is somewhat constraining, and as a result the table dependency has to be the other way around. Of course neither depends on each other's Java API.